pkgs/development/tools/godot/default.nix

```
@@ -1,108 +1,136 @@
-{ stdenv
-, lib
+{ alsa-lib
+, alsa-plugins
 , fetchFromGitHub
-, scons
-, pkg-config
-, udev
+, freetype
+, installShellFiles
+, lib
+, libGLU
+, libpulseaudio
 , libX11
 , libXcursor
+, libXext
+, libXfixes
+, libXi
 , libXinerama
 , libXrandr
 , libXrender
-, libpulseaudio
-, libXi
-, libXext
-, libXfixes
-, freetype
-, openssl
-, alsa-lib
-, alsa-plugins
 , makeWrapper
-, libGLU
-, zlib
+, openssl
+, pkg-config
+, scons
+, stdenv
+, udev
 , yasm
-, withUdev ? true
+, zlib
 }:
 
-let
-  options = {
-    touch = libXi != null;
-    pulseaudio = false;
-    udev = withUdev;
-  };
-in
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (self: {
   pname = "godot";
   version = "3.5.1";
 
   src = fetchFromGitHub {
     owner = "godotengine";
     repo = "godot";
-    rev = "${version}-stable";
+    rev = "${self.version}-stable";
     sha256 = "sha256-uHwTthyhfeQN0R1XjqZ+kGRa5WcpeQzA/DO9hZk4lvU=";
   };
 
-  nativeBuildInputs = [ pkg-config makeWrapper ];
-  buildInputs = [
+  nativeBuildInputs = [
+    installShellFiles
+    pkg-config
+    makeWrapper
     scons
-    udev
+  ];
+
+  buildInputs = [
+    alsa-lib
+    freetype
+    libGLU
+    libpulseaudio
     libX11
     libXcursor
+    libXext
+    libXfixes
+    libXi
     libXinerama
     libXrandr
     libXrender
-    libXi
-    libXext
-    libXfixes
-    freetype
     openssl
-    alsa-lib
-    libpulseaudio
-    libGLU
-    zlib
+    udev
     yasm
+    zlib
   ];
 
-  patches = [ ./pkg_config_additions.patch ./dont_clobber_environment.patch ];
+  patches = [ ./detect.py.patch ];
 
   enableParallelBuilding = true;
 
-  sconsFlags = "target=release_debug platform=x11";
-  preConfigure = ''
-    sconsFlags+=" ${
-      lib.concatStringsSep " "
-      (lib.mapAttrsToList (k: v: "${k}=${builtins.toJSON v}") options)
-    }"
-  '';
+  buildDescription = "x11 tools";
+  buildPlatform = "x11";
+  shouldBuildTools = true;
+  buildTarget = "release_debug";
 
-  outputs = [ "out" "dev" "man" ];
+  sconsFlags = [
+    "platform=${self.buildPlatform}"
+    "tools=${lib.boolToString self.shouldBuildTools}"
+    "target=${self.buildTarget}"
+    "bits=64"
+    "touch=true"
+    "pulseaudio=true"
+    "udev=true"
+  ];
+
+  shouldWrapBinary = self.shouldBuildTools;
+  shouldInstallHeaders = self.shouldBuildTools;
+  shouldInstallShortcut = self.shouldBuildTools && self.buildPlatform != "server";
+
+  outputs = ["out" "man"] ++ lib.optional self.shouldBuildTools "dev";
+
+  builtGodotBinNamePattern = if self.buildPlatform == "server" then "godot_server.*" else "godot.*";
+
+  godotBinInstallPath = "bin";
+  installedGodotBinName = self.pname;
 
   installPhase = ''
-    mkdir -p "$out/bin"
-    cp bin/godot.* $out/bin/godot
+    runHook preInstall
+
+    echo "Installing godot binaries."
+    outbin="$out/$godotBinInstallPath"
+    mkdir -p "$outbin"
+    cp -R bin/. "$outbin"
+    mv "$outbin"/$builtGodotBinNamePattern "$outbin/$installedGodotBinName"
+
+    if [ $shouldWrapBinary ]; then
```

This should be reflected everywhere the `if [ $var ]` pattern is used.

```suggestion
    if [ -n "$shouldWrapBinary" ]; then
```

---

Hmm, are you sure about that?  The `should*` variables are all booleans, and nix converts them to strings such that `true` becomes `"1"` and `false` becomes `""`.  In bash, if you do `if [ 1 ]; then ...; fi`, it's treated as true, and if you do `if [ ]; then ...; fi`, it's treated as false.  So it seems like the nix `toString` implementation for booleans is designed specifically to allow them to be used like this...

---

Didn't change this for the above reason.  Let me know if I'm wrong about something.